Zotero/Scenes/AppCoordinator.swift

```diff
@@ -258,7 +282,9 @@ final class AppCoordinator: NSObject {
                     case .local, .localAndChangedRemotely:
                         let file = Files.attachmentFile(in: libraryId, key: key, filename: filename, contentType: contentType)
                         url = file.createUrl()
-                        library = try coordinator.perform(request: ReadLibraryDbRequest(libraryId: libraryId))
+                        let (_collection, _library) = try coordinator.perform(request: ReadCollectionAndLibraryDbRequest(collectionId: collectionId, libraryId: libraryId))
```

Comment:
So `ReadCollectionAndLibraryDbRequest` returns `nil` for custom collections. I've checked and I think we can just change `ReadCollectionAndLibraryDbRequest` to return `Collection` for custom `CollectionIdentifier`s. So the result value can also be `(Collection, Library)` there too.

Reply:
Check https://github.com/zotero/zotero-ios/pull/718/commits/32026a7d276daa10e4c9408cc2411c435753002c